src/DBus.vala

```diff
@@ -103,34 +74,29 @@ public class Notifications.Server : Object {
         int32 expire_timeout,
         BusName sender
     ) throws DBusError, IOError {
+        // Silence "Automatic suspend. Suspending soon because of inactivity." notifications
+        // These values and hints are taken from gnome-settings-daemon source code
+        // See: https://gitlab.gnome.org/GNOME/gnome-settings-daemon/-/blob/master/plugins/power/gsd-power-manager.c#L356
+        // We must check for app_icon == "" to not block low power notifications
+        if ("desktop-entry" in hints && hints["desktop-entry"].get_string () == "gnome-power-panel"
+        && "urgency" in hints && hints["urgency"].get_byte () == 3
+        && app_icon == ""
+        && expire_timeout == 0
+        ) {
+            debug ("Blocked GSD notification");
+            throw new DBusError.FAILED ("Notification Blocked");
+        }
+
         var id = (replaces_id != 0 ? replaces_id : ++id_counter);
 
         if (hints.contains (X_CANONICAL_PRIVATE_SYNCHRONOUS)) {
             send_confirmation (app_icon, hints);
         } else {
             var notification = new Notifications.Notification (app_name, app_icon, summary, body, actions, hints);
-
-            // Silence "Automatic suspend. Suspending soon because of inactivity." notifications
-            // These values and hints are taken from gnome-settings-daemon source code
-            // See https://gitlab.gnome.org/GNOME/gnome-settings-daemon/-/blob/master/plugins/power/gsd-power-manager.c#L356
-            // We must check for app_icon == "" to not block low power notifications
-            if (hints.contains ("desktop-entry") && ((string) hints.get ("desktop-entry")) == "gnome-power-panel" &&
-                notification.priority == GLib.NotificationPriority.HIGH &&
-                app_icon == "" &&
-                expire_timeout == 0) {
-                debug ("Blocked GSD notification");
-                return id;
-            }
-
             if (!settings.get_boolean ("do-not-disturb") || notification.priority == GLib.NotificationPriority.URGENT) {
-                var app_id = notification.app_id;
-                if (notification.app_info == null || notification.app_info.get_boolean ("X-GNOME-UsesNotifications") == false) {
-                    app_id = "gala-other";
-                }
```

Comment:
This was added in https://github.com/elementary/notifications/pull/182 to fix https://github.com/elementary/wingpanel-indicator-notifications/issues/245. it there a reason to remove it?

Reply:
Oh sorry, just found it's checked properly in Notitfications.vala now. Sorry